tests/core/test_flow.py

```diff
@@ -1976,6 +1976,38 @@ def dummy_task():
         assert hashes[0]  # Ensure we don't have an empty string or None
         assert len(set(hashes)) == 1
 
+    @pytest.mark.skipif(
+        sys.version_info < (3, 8),
+        reason="Positional-Only arguments are only supported in Python 3.8+",
+    )
+    def test_task_positional_only_arguments(self, tmpdir):
+        contents = textwrap.dedent(
+            """
+        from prefect import task, Flow
+
+        @task
+        def dummy_task(a, b, /):
+            pass
+
+        with Flow("example-flow") as flow:
+            dummy_task()
+
+        if __name__ == "__main__":
+            flow.run()
+        """
+        )
+        script = tmpdir.join("flow.py")
+        script.write_text(contents, encoding="utf-8")
+
+        result = subprocess.run([sys.executable, str(script)], capture_output=True)
```

Comment:
The test above this uses a subprocess to run the flow because we're testing that a hash is consistent across Python instances. This test doesn't need a subprocess and they're at least an order of magnitude slower. Can you write the test with the flow/task definition inline and use `with pytest.raises` instead?

Reply:
Ah I see that makes a lot of sense. Can you just use `exec` instead of a subprocess then?